src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/on_way_point_change_notification_test.cc

```diff
@@ -99,11 +120,42 @@ MATCHER(CheckMessageData, "") {
 TEST_F(OnWayPointChangeNotificationTest,
        Run_NotEmptyListOfAppsSubscribedForWayPoints_SUCCESS) {
   std::set<int32_t> apps_subscribed_for_way_points;
-  apps_subscribed_for_way_points.insert(kAppId);
+  apps_subscribed_for_way_points.insert(kApp1Id);
 
   EXPECT_CALL(app_mngr_, GetAppsSubscribedForWayPoints())
       .WillOnce(Return(apps_subscribed_for_way_points));
-  EXPECT_CALL(mock_rpc_service_, SendMessageToMobile(CheckMessageData(), _));
+  EXPECT_CALL(mock_rpc_service_,
+              SendMessageToMobile(CheckMessageData(kApp1Id), _));
+
+  command_->Run();
+}
+
+TEST_F(OnWayPointChangeNotificationTest,
+       Run_StoreWayPointsDuringNotification_SUCCESS) {
+  std::set<int32_t> apps_subscribed_for_way_points;
+  apps_subscribed_for_way_points.insert(kApp1Id);
+
+  EXPECT_CALL(app_mngr_, GetAppsSubscribedForWayPoints())
+      .WillOnce(Return(apps_subscribed_for_way_points));
+  EXPECT_CALL(mock_rpc_service_,
+              SendMessageToMobile(CheckMessageData(kApp1Id), _));
+  EXPECT_CALL(app_mngr_, SaveWayPointsMessage(message_));
+
+  command_->Run();
```

Comment:
@ZhdanovP Where is command init() ? It should be always called before Run()

Reply:
@AByzhynar We don't overload the Init() method for `OnWayPointChangeNotification` , the initial implementation in `CommandNotificationImpl` just return `true` therefore the call this method here is redundant

Reply:
@AByzhynar I think it's better to add Init() call when it will have sense. But if you insist I will add.

Reply:
@ZhdanovP No. As command never should be used without initialization. If you go your way - in future after Init() would be changed - we have to go through entire list of commands and add everywhere Init() call.

Reply:
@AByzhynar Got it. I was also bewildered because we have a lot of places in SDL tests where the Init() don't call before Run() among them `unsubscribe_way_points_request_test` `send_location_request_test` and so on. But for this PR I will add. Thanks.